build-logic-commons/code-quality/src/main/kotlin/gradlebuild.code-quality.gradle.kts

```diff
@@ -41,7 +41,7 @@ val rules by configurations.creating {
     isCanBeConsumed = false
 
     attributes {
-        attribute(LibraryElements.LIBRARY_ELEMENTS_ATTRIBUTE, objects.named(LibraryElements.RESOURCES))
+        attribute(LibraryElements.LIBRARY_ELEMENTS_ATTRIBUTE, objects.named<LibraryElements>(LibraryElements.RESOURCES))
```

Comment:
Nevermind, I see you explained it in `upgrading_version_7.adoc`. Are we really sure Kotlin can't handle both argument types? Perhaps not, if the use-case is Object vs. Provider.

--

I wonder why we started parameterizing all the `objects.named` calls. I don't have a problem with it, I'm just curious.

Is it something to do with the erasure of `AttributeContainer#attribute(Attribute<T>, T)` conflicting with `public <T> AttributeContainer attribute(Attribute<T> key, Provider<? extends T> provider)` on `DefaultImmutableAtributes`?